rules/license-header.test.ts

```diff
@@ -63,23 +68,52 @@ var b = 2
 console.log(1 + 2)
 `;
 
-const invalidLichtblickHeaderCases = [
-  invalidLichtblickHeaderEmpty,
-  invalidLichtblickHeaderOlder,
-  invalidLichtblickHeaderRandom,
-];
+const invalidLichtblickHeaderWithMissingTypeOfLicense = `
+// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
+// SPDX-License-Identifier: ${noLicense}
+
+`;
 
 ruleTester.run("check-license-header", rule, {
   valid: [
-    validLichtblickHeader,
-    validLichtblickHeaderWithSpaces,
-    validLichtblickHeaderWithSpacesWithJsdom,
+    {
+      code: validLichtblickHeader,
+      options: [{ licenseType: "MPL-2.0" }],
+    },
+    {
+      code: validLichtblickHeaderWithSpaces,
+      options: [{ licenseType: "MPL-2.0" }],
+    },
+    {
+      code: validLichtblickHeaderWithSpacesWithJsdom,
+      options: [{ licenseType: "MPL-2.0" }],
+    },
   ],
 
   // Test if the lint fix were successfull, adding the LICENSE_HEADER followed by two empty lines
-  invalid: invalidLichtblickHeaderCases.map((invalidHeader) => ({
-    code: invalidHeader,
-    errors: [{ messageId: "missingLicenseError" }],
-    output: LICENSE_HEADER + "\n\n" + invalidHeader,
-  })),
+  invalid: [
+    {
+      code: invalidLichtblickHeaderEmpty,
+      options: [{ licenseType: "MPL-2.0" }],
+      errors: [{ messageId: "missingLicenseError" }],
+      output: LICENSE_HEADER + "\n\n" + invalidLichtblickHeaderEmpty,
+    },
+    {
+      code: invalidLichtblickHeaderOlder,
+      options: [{ licenseType: "MPL-2.0" }],
+      errors: [{ messageId: "missingLicenseError" }],
+      output: LICENSE_HEADER + "\n\n" + invalidLichtblickHeaderOlder,
+    },
+    {
+      code: invalidLichtblickHeaderRandom,
+      options: [{ licenseType: "MPL-2.0" }],
+      errors: [{ messageId: "missingLicenseError" }],
+      output: LICENSE_HEADER + "\n\n" + invalidLichtblickHeaderRandom,
+    },
+    {
+      code: invalidLichtblickHeaderWithMissingTypeOfLicense,
+      options: [],
+      errors: [{ messageId: "missingTypeOfLicense" }],
```

Comment:
if those errors are different why the messageId and licenseType are the same?

Reply:
The first three tests are testing to see if the right header exists and its at the beginning of the file, testing various hypothesis in that case, they all have the same licenseType because that isn't what I'm trying to test so I just kept the same license to be practical. On the last one I'm testing a different type of error, one checking the licenseType specifically, and not the header. 
The first three tests already existed I just had to include the license because of the changes I made on the rule